src/lib/checkers.js

```diff
@@ -1,17 +1,42 @@
 /* eslint-disable no-console */
+'use strict';
 
+/**
+ * Find users who currently belong to the organization, but not who are not
+ * expected by config.
+ *
+ * @param {string[]} configured - An array of usernames expected to exist in the org
+ * @param {string[]} retrieved - An array of usernames actually belong to the org currently
+ * @returns {string[]} An array of usernames which are not expected to belong to the org
+ */
 function findUndocumentedMembers(configured, retrieved) {
   const undocumented = retrieved.filter(member => !configured.includes(member));
   console.log(`${undocumented.length} undocumented memberships found: `, undocumented);
   return undocumented;
 }
 
+/**
+ * Find users who are expected by config, but who do not (yet) belong to the
+ * organization.
+ *
+ * @param {string[]} configured - An array of usernames expected to exist in the org
+ * @param {string[]} retrieved - An array of usernames actually belong to the org currently
+ * @returns {string[]} An array of usernames which are expected to belong to the org
+ */
 function findNewMembers(configured, retrieved) {
   const newMembers = configured.filter(member => !retrieved.includes(member));
   console.log(`${newMembers.length} new memberships requested: `, newMembers);
   return newMembers;
 }
 
+/**
+ * Find users who are currently admins of the org, but should not be, according
+ * to config.
+ *
+ * @param {object.<string, string>} configured - Configured usernames and their associated roles
+ * @param {object.<string, object.<string, string>>} retrieved - User list retrieved from GitHub
```

Comment:
IMHO you should make a [`@typedef`](https://jsdoc.app/tags-typedef.html) for these, then you can use that named typedef in all these functions where appropriate.

Reply:
I'll check that out, thanks! I did struggle for a good while on how to properly annotate these. Would that comment go at the top of this module? Is there a way to reference those typedefs across multiple modules (if I find the need to)?

Reply:
yeah, I'd just put all the typedefs at the top of the file.  I'm not sure if there's a good mechanism for annotating types across files, it might just work or not. There's some discussion of it [here](https://stackoverflow.com/questions/43183450/jsdoc-typedef-in-a-separate-file) with possible solutions.

Reply:
I added this:
```js
/**
 * A normalized member record retrieved from the GitHub GraphQL API
 *
 * @typedef {object} MemberRecord
 * @property {string} role - The user's role in the org; "ADMIN" or "MEMBER"
 * @property {boolean} hasTwoFactorEnabled - Whether the user has Two Factor Authentication enabled
 */

```

Reply:
I ended up going with the `typedefs.js` solution for cross-module references :)